src/elements/element.line.js

```diff
@@ -37,11 +37,24 @@ module.exports = Element.extend({
 		var globalDefaults = defaults.global;
 		var globalOptionLineElements = globalDefaults.elements.line;
 		var lastDrawnIndex = -1;
+		var closePath = me._loop;
 		var index, current, previous, currentVM;
 
-		// If we are looping, adding the first point again
 		if (me._loop && points.length) {
-			points.push(points[0]);
+			if (!spanGaps) {
+				for (index = points.length - 1; index >= 0; --index) {
+					// If the line has an open path, shift the point array
+					if (points[index]._view.skip) {
+						points = points.slice(index).concat(points.slice(0, index));
```

Comment:
Is this removing an element from the array? If so, I think `points.splice(index, 1)` may be an easier way to write that

Reply:
This is concatenating the [`index`, `points.length`) and [`0`, `index`). The both potions are arrays, so I think this is a shorter statement.

Reply:
Maybe expanding the comment with something saying ".. so line drawing start from and ends to that gap"

Reply:
@nagix What happen if the line has many gaps? Looks like the array is shifted multiple time but then I'm not sure to get the logic behind. Also it seems that every time there is a gap, the points array order is completely changed, thus `index` isn't anymore "valid" (even if iterated the reverse way).

Reply:
The array will be shifted one time at most when a gap is first detected by iterating the reverse way. One time is enough because it will guarantee that the line is joined at original index 0, and the number of gaps doesn't matter.

The `point` array is a cloned array and is only used in this function. So, I don't see any problem in changing it.

Reply:
Right, I didn't see the `break;`